consensus-types/types/deposit.go

```diff
@@ -54,6 +54,10 @@ type Deposit struct {
 	Index uint64 `json:"index"`
 }
 
+func NewEmptyDeposit() *Deposit {
+	return &Deposit{}
+}
```

Comment:
Needed for kvStore to allow them creating objects to unmarshal data into.
IMHO it's better to have a few of these small constructors, rather than making New a method of a generic interface